tck/src/main/java/org/eclipse/microprofile/lra/tck/TckTests.java

```diff
@@ -45,398 +36,508 @@
-        run.add("nestedActivity", TckTests::nestedActivity, verbose);
-        run.add("completeMultiLevelNestedActivity", TckTests::completeMultiLevelNestedActivity, verbose);
-        run.add("compensateMultiLevelNestedActivity", TckTests::compensateMultiLevelNestedActivity, verbose);
-        run.add("mixedMultiLevelNestedActivity", TckTests::mixedMultiLevelNestedActivity, verbose);
-        run.add("joinLRAViaHeader", TckTests::joinLRAViaHeader, verbose);
-        run.add("join", TckTests::join, verbose);
-        run.add("leaveLRA", TckTests::leaveLRA, verbose);
-        run.add("leaveLRAViaAPI", TckTests::leaveLRAViaAPI, verbose);
-        run.add("dependentLRA", TckTests::dependentLRA, verbose);
-        run.add("cancelOn", TckTests::cancelOn, verbose);
-        run.add("cancelOnFamily", TckTests::cancelOnFamily, verbose);
-        run.add("acceptTest", TckTests::acceptTest, verbose);
-        run.add("noLRATest", TckTests::noLRATest, verbose);
-
-        run.runTests(this, testname);
-
-        return run;
+    @Deployment(name = "tcktests", managed = true, testable = true)
+    public static WebArchive deploy() {
+        String archiveName = TckTests.class.getSimpleName().toLowerCase();
+        return ShrinkWrap
+            .create(WebArchive.class, archiveName + ".war")
+            .addPackages(true, "org.eclipse.microprofile.lra.tck")
+            .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
     }
-
-    private static void initTck(LRAClient lraClient, ManagementSPI managementSPI) {
-        TckTests.lraClient = lraClient;
-        TckTests.lraSPI = managementSPI;
-
-        try {
-            if (Boolean.valueOf(System.getProperty("enablePause", "true"))) {
-                System.out.println("Getting ready to connect - expecting swarm lra coordinator is already up...");
-                Thread.sleep(1000);
-            }
-
-            int servicePort = Integer.getInteger("service.http.port", TEST_SWARM_PORT);
-            // TODO issue 42 will be removing these endpoint references
-            String rcHost = System.getProperty(LRA_COORDINATOR_HOST_KEY, "localhost");
-            String rcPath = System.getProperty(LRA_RECOVERY_PATH_KEY, "lra-recovery-coordinator");
-            int rcPort = Integer.getInteger(LRA_COORDINATOR_PORT_KEY, COORDINATOR_SWARM_PORT);
-
-            micrserviceBaseUrl = new URL(String.format("http://localhost:%d", servicePort));
-            rcBaseUrl = new URL(String.format("http://%s:%d/%s", rcHost, rcPort, rcPath));
-
-            msClient = ClientBuilder.newClient();
-            rcClient = ClientBuilder.newClient();
-
-            oldLRAs = new ArrayList<>();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
-    }
-
+    
     @AfterClass
     public static void afterClass() {
-        oldLRAs.clear();
-        lraClient.close();
-        msClient.close();
-        rcClient.close();
+        if(tckSuiteClient != null) tckSuiteClient.close();
```

Comment:
Do we use checkstyle in the TCK?

Reply:
yes, the checkstyle is used. You mean that you don't like this without use of curly braces? 

```
if(tckSuiteClient != null)  {
    tckSuiteClient.close();
}
```

If you are about to add this rule to the checkstyle I can change the code here.

Reply:
I think that all projects should be using the same set of rules but I see that ours is different since we have commented out NeedBraces:

>  <!--module name="NeedBraces" / -->

I would recommend that that rule is re-enabled.

I will check to see where else we are different from other projects and if we are I will raise a separate issue for it (hopefully this is the only one).